src/servers/src/http.rs

```diff
@@ -104,6 +104,7 @@ pub struct HttpServer {
     script_handler: Option<ScriptHandlerRef>,
     shutdown_tx: Mutex<Option<Sender<()>>>,
     user_provider: Option<UserProviderRef>,
+    schema_validator: Option<SchemaValidatorRef>,
```

Comment:
Suggest providing a default SchemaValidator implementation that always returns true, to avoid using `Option` here, reducing some boilerplate None check or unwrap.

Same to `user_provider`.

Reply:
It's a little bit tricky here, the `Option` also serves as an indicator whether the server should enter authentication process, which may leads to some extra process like parsing http auth header or pg server returning a `AuthenticationInProgress` connection handshake. 

Otherwise a flag may be needed in server opts to explicitly tell if authentication process is required, which doesn't really save the checking?